ios/RNSScreenStackHeaderConfig.mm

```diff
@@ -624,6 +626,13 @@ + (void)updateViewController:(UIViewController *)vc
     }
   }
 
+  dispatch_async(dispatch_get_main_queue(), ^{
+    // Position the contents in the navigation bar, regarding to the direction.
+    for (UIView *view in navctr.navigationBar.subviews) {
+      view.semanticContentAttribute = config.direction;
+    }
+  });
```

Comment:
One comment regarding this line - Unfortunately I haven't managed to set `semanticContentAttribute` without the `dispatch_async` method. The problem is that:

 >iOS applies appearance changes when a view enters a window, it doesn’t change the appearance of a view >that’s already in a window. To change the appearance of a view that’s currently in a window, remove the view >from the view hierarchy and then put it back.

(via [SO](https://stackoverflow.com/questions/49277773/semanticcontentattribute-is-not-working) and Apple docs)

with is quite problematic for us, because we're setting attributes from config when the `ScreenStackHeaderConfig` is already mounted to the view, so dispatching it asynchronously is a good workaround for setting those content attributes.

Reply:
If so, can not we move the logic responsible for applying content attributes to some other place that would give us opportunity to set this earlier? 

> which is quite problematic for us, because we're setting attributes from config when the ScreenStackHeaderConfig is already mounted to the view, so dispatching it asynchronously is a good workaround for setting those content attributes.

IIRC `ScreenStackHeaderConfig` is not attached to view hierarchy at all on Paper. My guess is that the problem arises because the moment we receive header config (it is added by RN via `insertSubview:atIndex:` method (writing from memory here)) the owning `Screen` and it's view controller are already mounted into hierarchy. If it is really the case I see two options:

1. Hack it on JS side so that screen is aware of ltr/rtl direction (is it not already?) and set content attriubutes at the moment screen is inserted as the child of stack view by RN
2. Go with the workaround you suggested. But tbh I do not understand why it works

Reply:
I think we've discussed most of the things during the weekly meeting :D but I was unable to move the implementation of changing the position of elements inside `addSubview / insertReactSubview` 😕 

Moreover, I've discovered that what the documentation of `UIAppearance` says does not fit to our case 🤔 For changing the mode of elements to RTL the view of the navigation bar should be **already** mounted, so we need to apply the changes quickly after the view will appear - that's why I'm still standing for leaving the implementation of setting `semanticContentAttribute` in `dispatch_after` (previously this was `dispatch_async`, but I've changed it, as dispatch_async is dispatching the contents of its block immediately, whereas dispatch_after is executing it just after the block where its stays finishes its own execution).